x/profiles/keeper/handler.go

```diff
@@ -140,3 +149,88 @@ func handleMsgDeleteProfile(ctx sdk.Context, keeper Keeper, msg types.MsgDeleteP
 
 	return &result, nil
 }
+
+// handleMsgRequestDTagTransfer handles the request of a dTag transfer
+func handleMsgRequestDTagTransfer(ctx sdk.Context, keeper Keeper, msg types.MsgRequestDTagTransfer) (*sdk.Result, error) {
+	transferRequest := types.NewDTagTransferRequest(msg.CurrentOwner, msg.ReceivingUser)
+
+	if err := keeper.SaveDTagTransferRequest(ctx, transferRequest); err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+	}
+
+	ctx.EventManager().EmitEvent(sdk.NewEvent(
+		types.EventTypeDTagTransferRequest,
+		sdk.NewAttribute(types.AttributeCurrentOwner, transferRequest.CurrentOwner.String()),
+		sdk.NewAttribute(types.AttributeReceivingUser, transferRequest.ReceivingUser.String()),
+	))
+
+	result := sdk.Result{
+		Data:   keeper.Cdc.MustMarshalBinaryLengthPrefixed(transferRequest),
+		Events: ctx.EventManager().Events(),
+	}
+
+	return &result, nil
+}
+
+// handleMsgAcceptDTagTransfer handle the acceptance of a dTag transfer request
+func handleMsgAcceptDTagTransfer(ctx sdk.Context, keeper Keeper, msg types.MsgAcceptDTagTransfer) (*sdk.Result, error) {
+	requests := keeper.GetUserDTagTransferRequests(ctx, msg.CurrentOwner)
+
+	// Check if the receiving user request is present, if not return error
+	found := false
+	for _, req := range requests {
+		if req.ReceivingUser.Equals(msg.ReceivingUser) {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("no request made from %s", msg.ReceivingUser))
+	}
+
+	// Get the current owner profile
+	currentOwnerProfile, exist := keeper.GetProfile(ctx, msg.CurrentOwner)
+	if !exist {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("profile of %s doesn't exist", msg.CurrentOwner))
+	}
+
+	// Get the dTag to trade
+	dTagToTrade := currentOwnerProfile.DTag
+
+	// Save the current owner profile with his new dTag
+	currentOwnerProfile = currentOwnerProfile.WithDTag(msg.NewDTag)
+	err := keeper.SaveProfile(ctx, currentOwnerProfile)
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+	}
+
+	// check for an existent profile of the receiving user
+	receiverProfile, exist := keeper.GetProfile(ctx, msg.ReceivingUser)
+	if !exist {
+		receiverProfile = types.NewProfile(dTagToTrade, msg.ReceivingUser, ctx.BlockTime())
+	} else {
+		receiverProfile = receiverProfile.WithDTag(dTagToTrade)
+	}
+
+	err = keeper.SaveProfile(ctx, receiverProfile)
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+	}
+
+	keeper.DeleteAllDTagTransferRequests(ctx, msg.CurrentOwner)
+
+	ctx.EventManager().EmitEvent(sdk.NewEvent(
+		types.EventTypeDTagTransferAccept,
+		sdk.NewAttribute(types.AttributeCurrentOwner, msg.CurrentOwner.String()),
+		sdk.NewAttribute(types.AttributeReceivingUser, msg.ReceivingUser.String()),
```

Comment:
I think we can add a couple of attributes here like 

- the traded DTag
- the new DTag that the giver has chosen